src/frontend/src/common/queries/locations.ts

```diff
@@ -31,16 +54,29 @@ const fetchLocations = (): Promise<LocationsResponse> => {
   return getBackendApiJson(route);
 };
 
-export const USE_LOCATIONS_INFO_QUERY_KEY = {
+export const USE_PATHOGEN_DEPTH_LOCATIONS_INFO_QUERY_KEY = {
   entities: [ENTITIES.LOCATION_INFO],
-  id: "locationInfo",
+  id: "pathogenDepthocationInfo",
 };
 
-const ONE_HOUR = 1000 * 60 * 60; // for react-query; milliseconds
-
-export function useLocations(): UseQueryResult<LocationsResponse, unknown> {
-  return useQuery([USE_LOCATIONS_INFO_QUERY_KEY], fetchLocations, {
+export function useNamedPathogenDepthLocations(): UseQueryResult<
+  NamedLocationsResponse,
+  unknown
+> {
+  const state = store.getState();
+  const pathogen = selectCurrentPathogen(state);
+  // Choosing the query key based on the pathogen ensures that pathogens
+  // that use all of the locations for their trees don' have to re-fetch
+  // everything
+  const queryKey =
+    locationDepthPathogenConfig[pathogen] === null
+      ? USE_LOCATIONS_INFO_QUERY_KEY
+      : USE_PATHOGEN_DEPTH_LOCATIONS_INFO_QUERY_KEY;
```

Comment:
I don't know if it's possible to trigger this given how things are currently set up -- it's quite possible it never happens right now -- but I think this wouldn't actually fire a new fetch if we changed from pathogen to pathogen while the query was active somewhere in the component tree. Since `USE_PATHOGEN_DEPTH_LOCATIONS_INFO_QUERY_KEY` is a constant, both covid and mpox would see the same query key when firing `useNamedPathogenDepthLocations`. I think we'd need the query key to be lightly dynamic, and somehow also incorporate the pathogen slug or something so there's a difference in keys when we change pathogens.

Reply:
This bug really did occur.  To fix it, I moved setting the queryKey out to where this is called in the NSTreeModal.  That component seems to do a lot, but it does work.